master/buildbot/status/web/build.py

```diff
@@ -291,12 +305,15 @@ def content(self, req, cxt):
                 step['css_class'] = "not-started"
                 step['time_to_run'] = ""
 
-            cxt['steps'].append(step)
-
             step['link'] = path_to_step(req, s)
             step['text'] = " ".join(s.getText())
+
+            cxt['steps'].append(step)
+
+            if hasattr(s, 'step_type_obj') and  s.step_type_obj is Trigger:
```

Comment:
I am just thinking, that we can move `prepare_trigger_links` to Trigger class (and subclasses or base class). Then we can always invoke this method, but only Trigger do your job. What do you think about it?

Reply:
I move `prepare_trigger_links to` to BuildStepStatus. Trigger is a bad place because most method calls come from BuildStepStatus. 
If we move code to Trigger we will have sth like this `yield s.step_type_obj.add_subbuilds_urls(s)` (in my opinion very ugly)